### PR TITLE
Extended the coverage of existing defender tampering related rules

### PR DESCRIPTION
**Pull request by swachchhanda000**
### Summary of the Pull Request
Extended the coverage of existing defender tampering related rules

### Detailed Description of the Pull Request / Additional Comments
The existing defender tampering related rules has lot of false-negative. So, rules to detect different techniques used for tampering windows defender are added.

### Example Log Event
N/A

### Fixed Issues
N/A

### SigmaHQ Rule Creation Conventions

- If your PR adds new rules, please consider following and applying these [conventions](https://github.com/SigmaHQ/sigma-specification/blob/main/sigmahq/sigmahq_conventions.md)

### CODE REVIEW

**Comment by nasbench** [comment]
@swachchhanda000 no need for multiple fixes. I'll check the PR later as I will apply additional changes :)

**Comment by swachchhanda000** [comment]
image
Uploading this image as it could be helpful

**Comment by swachchhanda000** [comment]
Hi @nasbench,
During one of the checks, I got the error that goes like
```
{"RulePath":"rules/windows/registry/registry_set/registry_set_windows_defender_tamper.yml","RuleTitle":"Disable Windows Defender Functionalities Via Registry Keys","RuleId":"0eb46774-f1ab-4a74-8238-1155855f2263","RuleLevel":"high","MatchStrings":[{"SearchIdentifier":"selection_main","Field":"EventType","Offset":0,"Data":"SetValue"},{"SearchIdentifier":"selection_main","Field":"TargetObject","Offset":4,"Data":"\\SOFTWARE\\Microsoft\\Windows Defender\\"},{"SearchIdentifier":"selection_dword_0","Field":"TargetObject","Offset":40,"Data":"\\Features\\TamperProtection"},{"SearchIdentifier":"selection_dword_0","Field":"Details","Offset":0,"Data":"DWORD (0x00000000)"}],"Event":"RuleName: -  EventType: SetValue  UtcTime: 2022-04-04 13:33:17.641  ProcessGuid: CCEE75F4-EE75-624A-2D00-000000000500  ProcessId: 1796  Image: C:\\ProgramData\\Microsoft\\Windows Defender\\Platform\\4.18.2202.4-0\\MsMpEng.exe  TargetObject: HKLM\\SOFTWARE\\Microsoft\\Windows Defender\\Features\\TamperProtection  Details: DWORD (0x00000000)  User: NT AUTHORITY\\SYSTEM  Provider_Name: Microsoft-Windows-Sysmon  Provider_Guid: 5770385F-C22A-43E0-BF4C-06F5698FFBD9  EventID: 13  Version: 2  Level: 4  Task: 13  Opcode: 0  Keywords: 9223372036854775808  TimeCreated_SystemTime: 1.6490791977867775e+09  EventRecordID: 445110  Execution_ProcessID: 1720  Execution_ThreadID: 2732  Channel: Microsoft-Windows-Sysmon/Operational  Computer: WIN-TKC15D7KHUR  Security_UserID: S-1-5-18  EventID: 13  ","File":"win2022-evtx/Microsoft-Windows-Sysmon%4Operational.evtx","Channel":"Microsoft-Windows-Sysmon/Operational"}
      2 {"RuleId":"0eb46774-f1ab-4a74-8238-1155855f2263","RuleTitle":"Disable Windows Defender Functionalities Via Registry Keys","RuleLevel":"high"}
      You either need to tune your rule(s) for false positives or add a false positive filter to .github/workflows/known-FPs.csv
  ```
   What was that about if you could?

**Comment by nasbench** [comment]
> Hi @nasbench, During one of the checks, I got the error that goes like
> 
> ```
> {"RulePath":"rules/windows/registry/registry_set/registry_set_windows_defender_tamper.yml","RuleTitle":"Disable Windows Defender Functionalities Via Registry Keys","RuleId":"0eb46774-f1ab-4a74-8238-1155855f2263","RuleLevel":"high","MatchStrings":[{"SearchIdentifier":"selection_main","Field":"EventType","Offset":0,"Data":"SetValue"},{"SearchIdentifier":"selection_main","Field":"TargetObject","Offset":4,"Data":"\\SOFTWARE\\Microsoft\\Windows Defender\\"},{"SearchIdentifier":"selection_dword_0","Field":"TargetObject","Offset":40,"Data":"\\Features\\TamperProtection"},{"SearchIdentifier":"selection_dword_0","Field":"Details","Offset":0,"Data":"DWORD (0x00000000)"}],"Event":"RuleName: -  EventType: SetValue  UtcTime: 2022-04-04 13:33:17.641  ProcessGuid: CCEE75F4-EE75-624A-2D00-000000000500  ProcessId: 1796  Image: C:\\ProgramData\\Microsoft\\Windows Defender\\Platform\\4.18.2202.4-0\\MsMpEng.exe  TargetObject: HKLM\\SOFTWARE\\Microsoft\\Windows Defender\\Features\\TamperProtection  Details: DWORD (0x00000000)  User: NT AUTHORITY\\SYSTEM  Provider_Name: Microsoft-Windows-Sysmon  Provider_Guid: 5770385F-C22A-43E0-BF4C-06F5698FFBD9  EventID: 13  Version: 2  Level: 4  Task: 13  Opcode: 0  Keywords: 9223372036854775808  TimeCreated_SystemTime: 1.6490791977867775e+09  EventRecordID: 445110  Execution_ProcessID: 1720  Execution_ThreadID: 2732  Channel: Microsoft-Windows-Sysmon/Operational  Computer: WIN-TKC15D7KHUR  Security_UserID: S-1-5-18  EventID: 13  ","File":"win2022-evtx/Microsoft-Windows-Sysmon%4Operational.evtx","Channel":"Microsoft-Windows-Sysmon/Operational"}
>       2 {"RuleId":"0eb46774-f1ab-4a74-8238-1155855f2263","RuleTitle":"Disable Windows Defender Functionalities Via Registry Keys","RuleLevel":"high"}
>       You either need to tune your rule(s) for false positives or add a false positive filter to .github/workflows/known-FPs.csv
> ```
> 
> What was that about if you could?

This means that the rule matches with the baseline checks we have. Which indicate that this happens during normal system activity and needs to be tuned or either whitelisted.

Basically don't worry about it :)

**Comment by swachchhanda000** [comment]
> > Hi @nasbench, During one of the checks, I got the error that goes like
> > ```
> > {"RulePath":"rules/windows/registry/registry_set/registry_set_windows_defender_tamper.yml","RuleTitle":"Disable Windows Defender Functionalities Via Registry Keys","RuleId":"0eb46774-f1ab-4a74-8238-1155855f2263","RuleLevel":"high","MatchStrings":[{"SearchIdentifier":"selection_main","Field":"EventType","Offset":0,"Data":"SetValue"},{"SearchIdentifier":"selection_main","Field":"TargetObject","Offset":4,"Data":"\\SOFTWARE\\Microsoft\\Windows Defender\\"},{"SearchIdentifier":"selection_dword_0","Field":"TargetObject","Offset":40,"Data":"\\Features\\TamperProtection"},{"SearchIdentifier":"selection_dword_0","Field":"Details","Offset":0,"Data":"DWORD (0x00000000)"}],"Event":"RuleName: -  EventType: SetValue  UtcTime: 2022-04-04 13:33:17.641  ProcessGuid: CCEE75F4-EE75-624A-2D00-000000000500  ProcessId: 1796  Image: C:\\ProgramData\\Microsoft\\Windows Defender\\Platform\\4.18.2202.4-0\\MsMpEng.exe  TargetObject: HKLM\\SOFTWARE\\Microsoft\\Windows Defender\\Features\\TamperProtection  Details: DWORD (0x00000000)  User: NT AUTHORITY\\SYSTEM  Provider_Name: Microsoft-Windows-Sysmon  Provider_Guid: 5770385F-C22A-43E0-BF4C-06F5698FFBD9  EventID: 13  Version: 2  Level: 4  Task: 13  Opcode: 0  Keywords: 9223372036854775808  TimeCreated_SystemTime: 1.6490791977867775e+09  EventRecordID: 445110  Execution_ProcessID: 1720  Execution_ThreadID: 2732  Channel: Microsoft-Windows-Sysmon/Operational  Computer: WIN-TKC15D7KHUR  Security_UserID: S-1-5-18  EventID: 13  ","File":"win2022-evtx/Microsoft-Windows-Sysmon%4Operational.evtx","Channel":"Microsoft-Windows-Sysmon/Operational"}
> >       2 {"RuleId":"0eb46774-f1ab-4a74-8238-1155855f2263","RuleTitle":"Disable Windows Defender Functionalities Via Registry Keys","RuleLevel":"high"}
> >       You either need to tune your rule(s) for false positives or add a false positive filter to .github/workflows/known-FPs.csv
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > What was that about if you could?
> 
> This means that the rule matches with the baseline checks we have. Which indicate that this happens during normal system activity and needs to be tuned or either whitelisted.
> 
> Basically don't worry about it :)

**Comment by nasbench** [comment]
@swachchhanda000 just to fully explain. Basically these test in the SS

image

Are test against legitimates EVTX that every rules need to pass for the PR to be merged. If something matches, this means that the rule need some tuning as it's matching on legit system / typical user activity

Cheers.

File: rules/windows/process_creation/proc_creation_win_reg_defender_tampering.yml [hunk 1]
```diff
@@ -22,18 +23,38 @@ detection:
         CommandLine|contains:
             - 'SOFTWARE\Microsoft\Windows Defender\'
             - 'SOFTWARE\Policies\Microsoft\Windows Defender\'
+            - 'SOFTWARE\Policies\Microsoft\Windows Defender Security Center' 
     selection_dword_0:
         CommandLine|contains|all:
             - ' add '
             - ' /d 0'
+            - ' -d 0'
```
Comment (frack113) [inline rules/windows/process_creation/proc_creation_win_reg_defender_tampering.yml:31]: caution it is a `|all` selection so the nex detection is for ` add /d 0 -d 0`

Comment (nasbench) [reply rules/windows/process_creation/proc_creation_win_reg_defender_tampering.yml:31]: Yes. that's why I said i'll take over this PR and fix some stuff :)

**Review by nasbench — APPROVED** [review]

File: rules/windows/powershell/powershell_classic/posh_pc_tamper_windows_defender_set_mp.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
+title: Tamper Windows Defender - PSClassic
+id: ec19ebab-72dc-40e1-9728-4c0b805d722c
+related:
+    - id: ec19ebab-72dc-40e1-9728-4c0b805d722c
+      type: similar
+status: experimental
+description: Attempting to disable scheduled scanning and other parts of windows defender atp. Or set default actions to allow.
```
Comment (phantinuss) [inline rules/windows/powershell/powershell_classic/posh_pc_tamper_windows_defender_set_mp.yml:7]:
```suggestion
description: Attempting to disable scheduled scanning and other parts of Windows Defender ATP or set default actions to allow.
```

File: rules/windows/powershell/powershell_classic/posh_pc_tamper_windows_defender_set_mp.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+author: frack113, Nasreddine Bencherchali (Nextron Systems)
+date: 2021/06/07
+modified: 2023/06/05
+tags:
+    - attack.defense_evasion
+    - attack.t1562.001
+logsource:
+    product: windows
+    category: ps_classic_provider_start
+    definition: fields have to be extract from event
```
Comment (phantinuss) [inline rules/windows/powershell/powershell_classic/posh_pc_tamper_windows_defender_set_mp.yml:19]:
```suggestion
    definition: fields have to be extracted from the event
```

File: rules/windows/process_creation/proc_creation_win_reg_disable_sec_services.yml [hunk 4]
```diff
@@ -1,15 +1,15 @@
-title: Reg Disable Security Service
+title: Security Service Disabled Via Reg.EXE
 id: 5e95028c-5229-4214-afae-d653d573d0ec
-status: experimental
-description: Detects a suspicious reg.exe invocation that looks as if it would disable an important security service
+status: test
+description: Detects execution of "reg.exe" to disable security service such as Windows Defender.
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_reg_disable_sec_services.yml:4]:
```suggestion
description: Detects execution of "reg.exe" to disable security services such as Windows Defender.
```

File: rules/windows/process_creation/proc_creation_win_reg_windows_defender_tamper.yml [hunk 5]
```diff
@@ -0,0 +1,67 @@
+title: Suspicious Windows Defender Registry Key Tampering Via Reg.EXE
+id: 452bce90-6fb0-43cc-97a5-affc283139b3
+status: experimental
+description: Detects the usage of "reg.exe" to tamper with different Windows defender registry keys in order to disable some important features related to protection and detection
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_reg_windows_defender_tamper.yml:4]:
```suggestion
description: Detects the usage of "reg.exe" to tamper with different Windows Defender registry keys in order to disable some important features related to protection and detection
```

**Review by phantinuss — APPROVED** [review]

**Pull request merged**